geopandas/geodataframe.py

```diff
@@ -214,10 +214,33 @@ def __setattr__(self, attr, val):
 
     def _get_geometry(self):
         if self._geometry_column_name not in self:
-            raise AttributeError(
-                "No geometry data set yet (expected in"
-                " column '%s'.)" % self._geometry_column_name
-            )
+            if self._geometry_column_name is None:
+                msg = (
+                    "You are calling a geospatial method on the GeoDataFrame, "
+                    "but the default active geometry column to use has not been set. "
+                )
+            else:
+                msg = (
+                    "You are calling a geospatial method on the GeoDataFrame, "
+                    f"but the active geometry column ({self._geometry_column_name}) "
+                    "is no longer present. "
```

Comment:
```suggestion
                    f"but the active geometry column ('{self._geometry_column_name}') "
                    "is not present. "
```
Wrapping string name in `'`. And removing "no longer" as it might have never been there.